contrib/buildrpm/buildrpmLibfabric.sh

```diff
@@ -0,0 +1,368 @@
+====              Welcome to Libfabric RPM build script                   ====
+------------------------------------------------------------------------------
+You have specified this options:
+ $arguments
+------------------------------------------------------------------------------
+
+EOF
+  if [[ -n "$noop" ]]; then
+    echo "$0: --> Option -n was specified. I won't do anything."
+  fi
+fi
+
+################################
+# Check if tarball was specified
+################################
+export tarball="$1"
+
+if [[ ! -f "$tarball" ]]; then
+  echo "$0: FATAL: Can't find $tarball" 1>&2
+  exit 2
+fi
+if [[ -n "$verbose" ]]; then
+  echo "$0: --> Found tarball: $tarball"
+fi
+
+##############################
+# Get tarball name and version
+##############################
+export tardirname=$(basename "$tarball" | awk '{print substr ($0, 0, index($0, ".tar")-1)}')
+if [[ -z "$tardirname" ]]; then
+  echo "$0: FATAL: Cannot determine name from $tarball" 1>&2
+  exit 3
+fi
+if [[ -n "$verbose" ]]; then
+  echo "$0: --> Name of package: $tarball"
+fi
+
+export version=$(basename "$tardirname" | cut -d- -f2)
+if [[ -z "$tardirname" ]]; then
+  echo "$0: FATAL: Cannot determine version from $tarball" 1>&2
+  exit 4
+fi
+if [[ -n "$verbose" ]]; then
+  echo "$0: --> Version: $version"
+fi
+
+if [[ -n "$install_in_opt" ]]; then
+  prefix="/opt/libfabric/$version"
+  rpmbuild_options="$rpmbuild_options --define '_prefix $prefix'"
+  if [[ -n "$verbose" ]]; then
+    echo "$0: --> Setting RPM install path to: $prefix"
+  fi
+fi
+
+######################################
+# Try to unpack spec file from tarball
+######################################
+export tar_cmd=""
+export cp_cmd=""
+export rm_cmd=""
+
+if [[ -f "$specfile" ]]; then
+  if [[ -n "$verbose" ]]; then
+    echo "$0: WARNING: $specfile already exists and will be overwritten" 1>&2
+  fi
+fi
+
+if [[ -n "$unpack_spec" ]]; then
+  if [[ -z "$verbose" ]]; then
+    if [[ -z "$noop" ]]; then
+      tar -xf "$tarball"
+      cp -fp "$tardirname/$specfile" .
+      rm -rf "$tardirname"
+    fi
+  else
+    tar_cmd="tar -xvf \"$tarball\""
+    cp_cmd="cp -fpv \"$tardirname/$specfile\" ."
+    rm_cmd="rm -rfv \"$tardirname\""
+    if [[ -z "$noop" ]]; then
+      echo "$0: --> Extracting tarball"
+      eval "$tar_cmd"
+      echo "$0: --> Copying specfile"
+      eval "$cp_cmd"
+      echo "$0: --> Cleanup after extraction"
+      eval "$rm_cmd"
+    else
+      echo "$0: --> Extracting tarball using command:"
+      echo "$0: --> $tar_cmd"
+      echo "$0: --> Copying specfile with command:"
+      echo "$0: --> $cp_cmd"
+      echo "$0: --> Cleanup after extraction with command:"
+      echo "$0: --> $rm_cmd"
+    fi
+  fi
+fi
+
+##############################
+# Check if specfile is present
+##############################
+if [[ ! -f $specfile ]]; then
+  echo "$0: FATAL: Cannot find $specfile" 1>&2
+  if [[ -z "$noop" ]]; then
+    exit 5
```

Comment:
Ah, I see.

But it's still weird/confusing: if you're in no-op mode and you unpack the specfile from the tarball, then you'll still see a misleading / confusing "FATAL: ..." message during the output.

You might want to re-think that sequence a bit to make sure that the output is correct / clear in both the no-op and non-no-op cases.